include/swift/Reflection/ReflectionContext.h

```diff
@@ -644,6 +645,46 @@ class ReflectionContext
     return false;
   }
 
+  /// Adds an image using the FindSection closure to find the swift metadata
+  /// sections. \param[in] FindSection
+  ///     Closure that finds sections by name. ReflectionContext is in charge
+  ///     of freeing the memory buffer in the RemoteRef return value.
+  ///     process.
+  /// \return
+  ///     /b True if any of the reflection sections were registered,
+  ///     /b false otherwise.
+  bool
+  addImage(std::function<std::pair<RemoteRef<void>, uint64_t>(llvm::StringRef)>
+               FindSection) {
+    auto &ObjectFileFormat = SwiftObjectFileFormat::getSwiftObjectFileFormat();
+    auto Sections = {
+        ReflectionSectionKind::fieldmd, ReflectionSectionKind::assocty,
+        ReflectionSectionKind::builtin, ReflectionSectionKind::capture,
+        ReflectionSectionKind::typeref, ReflectionSectionKind::reflstr};
+
+    std::vector<std::pair<RemoteRef<void>, uint64_t>> Pairs;
+    for (auto Section : Sections) {
+      Pairs.push_back(FindSection(ObjectFileFormat.getSectionName(Section)));
+      if (Pairs.back().first) {
```

Comment:
since we use `Pairs.back().first` multiple times and it's not obvious what it is, perhaps write this as `auto descriptive_name = Pairs.back().first;`?